html/user/submit_rpc_handler.php

```diff
@@ -644,6 +644,32 @@ function show_job_details($wu) {
     }
 }
 
+function show_job_status($wu) {
+    if       ($wu->error_mask) {
+        echo "            <status>error</status>\n";
+    } elseif ($wu->canonical_resultid) {
+        echo "            <status>done</status>\n";
+    } else {
+        $results = BoincResult::enum("server_state","workunitid=$wu->id");
+        $in_progress = 0;
+        foreach ($results as $r) {
```

Comment:
There is no need to iterate here. This shows the code really is copy-paste from the function above.

This could be done in SQL, "workunitid=$wu->id AND server_state=".RESULT_SERVER_STATE_IN_PROGRESS - and then check if the array size is > 0.